creator-node/src/middlewares/premiumContent/premiumContentMiddleware.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  sendResponse,
+  errorResponseServerError,
+  errorResponseForbidden,
+  errorResponseBadRequest
+} from '../../apiHelpers'
+import { recoverWallet } from '../../apiSigning'
+import { NextFunction, Request, Response } from 'express'
+import { isPremiumContentMatch } from '../../premiumContent/helpers'
+import { PremiumContentType } from '../../premiumContent/types'
+import { getAllRegisteredDNodes } from '../../utils'
+import type Logger from 'bunyan'
+import { Redis } from 'ioredis'
+
+/**
+ * Middleware to validate requests to get premium content.
+ */
+export const premiumContentMiddleware = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const {
+      signedDataFromDiscoveryNode,
+      signatureFromDiscoveryNode,
+      signedDataFromUser,
+      signatureFromUser
+    } = req.headers
+
+    if (
+      !signedDataFromDiscoveryNode ||
+      !signatureFromDiscoveryNode ||
+      !signedDataFromUser ||
+      !signatureFromUser
+    ) {
+      return sendResponse(
+        req,
+        res,
+        errorResponseBadRequest('Missing request headers.')
+      )
+    }
+
+    const logger = (req as any).logger as Logger
+
+    const discoveryNodeWallet = recoverWallet(
+      signedDataFromDiscoveryNode,
+      signatureFromDiscoveryNode
+    )
+    const isRegisteredDN = await isRegisteredDiscoveryNode({
+      wallet: discoveryNodeWallet,
+      libs: req.app.get('audiusLibs'),
+      logger,
+      redis: req.app.get('redisClient')
+    })
+    if (!isRegisteredDN) {
+      return sendResponse(
+        req,
+        res,
+        errorResponseForbidden(
+          'Failed discovery node signature validation for premium content.'
+        )
+      )
+    }
+
+    const { premiumContentId, premiumContentType } = req.params
+
+    const userWallet = recoverWallet(signedDataFromUser, signatureFromUser)
+    if (
+      !isPremiumContentMatch({
```

Comment:
Should probably unpack this line – big function call within a conditional including a nested JSON.parse feels like a bit of a smell